### DISCUSSION TITLE
yarn.lock must be exist in the project?

### DISCUSSION THREAD

**Discussion by ragingwind** [vercel/turborepo#597] [Help] [answered]
Run `turbo run build` command return error `ERROR  yarn.lock: failed to hash lockfile: open yarn.lock: no such file or directory` even though npm was chosen as default package manager while creating a project. Following [the code](https://github.com/vercel/turborepo/blob/6a5d70e416ac40ccd8377a5e9437f734cfd365df/cli/internal/fs/lockfile.go#L14), yarn.lock should be required.

**Comment by samchouse** [comment, marked as answer]
Do you a specfile in your project directory eg: `package-lock.json`. If not Yarn is selected by default. Run the install command with your package manager and everything should be fixed.

Edit: if a specfile does exist make sure you're running the command in the root directory of the monorepo.
